models/endoscopic_inbody_classification/configs/multi_gpu_train.json

```diff
@@ -34,6 +34,6 @@
         "$@train#trainer.run()"
     ],
     "finalize": [
-        "$dist.destroy_process_group()"
+        "$dist.is_initialized() and dist.destroy_process_group()"
```

Comment:
Do we need 'dist.is_available() and dist.is_initialized()'?

Reply:
Hi @wyli , thanks for the suggestion. It seems `dist.is_available()` is always `True` if the distributed package is avaiable. I'm not sure if we need to add it. Correct me if I misunderstand anything here, thanks!

Reply:
if we don't check it here, I think we'll have issue like this one at some point https://github.com/facebookresearch/maskrcnn-benchmark/issues/280

Reply:
Hi @wyli ,

I think this PR is mainly to avoid duplicated "destroy dist" operations if launching several train workflow and evaluate workflow together.
If the user's program can run until this "finalize" section, at least the `dist` should be OK? So maybe no need to add the "dist.is_available()".

Thanks.

Reply:
from the python API, now there's the possibility that `finalize` getting called at any time, correct?

Reply:
it might be an issue for the initialize as well

Reply:
Hi @wyli ,
> there's the possibility that finalize getting called at any time

that's true and if not having the distributed package, use `dist.is_available() and dist.is_initialized()` will not raise any error.
However, in this case I think the error is informative and useful (can let users know ddp is not suitable). What do you think?

Reply:
The error will be AttributeError: module 'torch.distributed' has no attribute 'is_initialized'

i don't think its informative.

Reply:
Hi @wyli ,

I understand your point.
If using `dist.is_available() and dist.is_initialized()`, it will not raise any error, how to raise a proper error?
We didn't find a better expression, do you have any suggestion for this expression?
CC @holgerroth .

Thanks in advance.

Reply:
I think it's fine to not raising an error in `finalize`. you can consider a better error message for the `initialize` section. maybe it's out of scope of this PR though.

Reply:
ok sure, we spend too much time on this. please feel free to ignore my comments and merge this.